ui/src/app/applications/components/application-node-info/application-node-info.tsx

```diff
@@ -16,10 +17,70 @@ import {
     getPodStateReason,
     HealthStatusIcon
 } from '../utils';
-
 import './application-node-info.scss';
 import {ReadinessGatesFailedWarning} from './readiness-gates-failed-warning';
 
+const RenderContainerState = (props: {container: any}) => {
+    const state = (props.container.state?.waiting && 'waiting') || (props.container.state?.terminated && 'terminated') || (props.container.state?.running && 'running');
+    const status = props.container.state.waiting?.reason || props.container.state.terminated?.reason || props.container.state.running?.reason;
+    const lastState = props.container.lastState?.terminated;
+    const msgExists = props.container.state.waiting?.message || props.container.state.terminated?.message;
+    return (
+        <div className='container'>
+            <div className='container__name'>{props.container.name}</div>
+            <div>
+                {state && (
+                    <>
+                        Container is <span className='application-node-info__labels--highlight'>{state}</span>
+                        {state !== 'running' && ' because of '}
+                    </>
+                )}
+                <span title={msgExists || ''}>
+                    {status && (
+                        <span
+                            className={classNames('application-node-info__labels--highlight', {
+                                'application-node-info__labels--hint': !!msgExists
+                            })}>
+                            {status}
+                        </span>
+                    )}
+                </span>
+                {'.'}
+                {(props.container.state.terminated?.exitCode === 0 || props.container.state.terminated?.exitCode) && (
+                    <>
+                        {' '}
+                        It exited with <span className='application-node-info__labels--highlight'>exitCode {props.container.state.terminated.exitCode}.</span>
+                    </>
+                )}
+                <>
+                    {' '}
+                    It is <span className='application-node-info__labels--highlight'>{props.container?.started ? 'started' : 'not started'}</span> and
+                    <span className='application-node-info__labels--highlight'>{props.container?.ready ? ' ready.' : ' not ready.'}</span>
+                </>
+                <br />
+                {lastState && (
+                    <>
+                        <>
+                            The container last terminated with <span className='application-node-info__labels--highlight'>exit code {lastState?.exitCode}</span> because of{' '}
```

Comment:
Should `because of{' '}` be part of the `reason` block below?

Reply:
Yup, Did the change.